components/compliance-service/profiles/db/store.go

```diff
@@ -376,89 +379,108 @@ func (s *Store) Delete(namespace string, name string, version string) error {
 		if err != nil {
-			return nil, err
+			return nil, 0, err
 		}
 		logrus.Debug("parse metadata")
 		metadata := inspec.Metadata{}
 		err = metadata.ParseJSON(metadataBlob)
 		if err != nil {
-			return nil, err
+			return nil, 0, err
 		}
 		metadata.Sha256 = sha256
 		entries = append(entries, metadata)
 	}
 	err := rows.Err()
 	if err != nil {
-		return nil, err
+		return nil, 0, err
 	}
-	return entries, nil
+	return entries, total, nil
 }
 
-func (s *Store) ListProfilesMetadata(namespace string, name string, sort string, order string) ([]inspec.Metadata, error) {
-	if order != "ASC" && order != "DESC" {
-		return nil, status.Errorf(codes.InvalidArgument, "order field '%s' is invalid. Use either 'ASC' or 'DESC'", order)
-	}
-
-	if sort != "name" && sort != "title" && sort != "maintainer" {
-		return nil, status.Errorf(codes.InvalidArgument, "sort field '%s' is invalid. Use either 'name', 'title' or 'maintainer'", sort)
-	}
-
-	var rows *sql.Rows
-	var err error
-
-	switch {
-	case len(namespace) == 0 && len(name) == 0:
-		query := fmt.Sprintf(`
-			SELECT sha256, info
-			FROM store_profiles WHERE sha256 IN
-			(SELECT sha256 FROM store_market)
-			ORDER BY store_profiles.info #>> '{%s}' %s, store_profiles.info #>> '{version}' DESC;
-		`, sort, order)
-
-		rows, err = s.DB.Query(query)
-	case len(namespace) == 0:
-		query := fmt.Sprintf(`
-			SELECT sha256, info
-			FROM store_profiles WHERE sha256 IN
-			(SELECT sha256 FROM store_market)
-			AND info->>'name' = $1
-			ORDER BY store_profiles.info #>> '{%s}' %s, store_profiles.info #>> '{version}' DESC;
-		`, sort, order)
-
-		rows, err = s.DB.Query(query, name)
-	case len(name) == 0:
-		query := fmt.Sprintf(`
-			SELECT sha256, info
-			FROM store_profiles WHERE
-			sha256 IN (SELECT sha256 FROM store_namespace WHERE owner=$1)
-			ORDER BY store_profiles.info #>> '{%s}' %s, store_profiles.info #>> '{version}' DESC;
-		`, sort, order)
-
-		rows, err = s.DB.Query(query, namespace)
-	default:
-		query := fmt.Sprintf(`
-			SELECT sha256, info
-			FROM store_profiles WHERE
-			sha256 IN (SELECT sha256 FROM store_namespace WHERE owner=$1)
-			AND info->>'name' = $2
-			ORDER BY store_profiles.info #>> '{%s}' %s, store_profiles.info #>> '{version}' DESC;
-		`, sort, order)
-
-		rows, err = s.DB.Query(query, namespace, name)
+type ProfilesListRequest struct {
+	Filters   map[string][]string
+	Name      string
+	Namespace string
+	Order     string
+	Page      int
+	PerPage   int
+	Sort      string
+}
+
+func (s *Store) ListProfilesMetadata(req ProfilesListRequest) ([]inspec.Metadata, int, error) {
+	if req.Order != "ASC" && req.Order != "DESC" {
+		return nil, 0, status.Errorf(codes.InvalidArgument, "order field '%s' is invalid. Use either 'ASC' or 'DESC'", req.Order)
+	}
+
+	if req.Sort != "name" && req.Sort != "title" && req.Sort != "maintainer" {
+		return nil, 0, status.Errorf(codes.InvalidArgument, "sort field '%s' is invalid. Use either 'name', 'title' or 'maintainer'", req.Sort)
```

Comment:
maintainer?  should that be namespace or owner?

Reply:
Wow, yeah, that's twice now I've blindly copied and pasted this code without noticing that! Maybe this table used to have that column?

Not everything has an owner so it would be somewhat complicated to add sorting on that. I'll just delete it.

Reply:
Now that I've actually looked at the data, there is a maintainer field included in the json document, so I'll just leave it. This does affect the followup PR though so thank you for pointing it out!

Reply:
maintainer comes in on the inspec yml for the profile -- if it means more work, i would default to not supporting that field for sort b/c it's not really a popular one/only shown on profile detail pages